HELPS/ClientApp/src/components/user/User.tsx

```diff
@@ -19,6 +31,14 @@ class User extends Component<UserProps> {
     }
 
     render() {
+        let _loadAllUsers = this.state.loadAllUsers;
+        let Search = this.state.searchTerm;
+
+        if (Search.length > 0) {
+            _loadAllUsers = _loadAllUsers.filter(function (loadAllUsers) {
```

Comment:
This is good but should be filtering props.student (I would rename to props.students) rather that loadAllusers

Also, filtering on student ID is fine but I would filter on whatever value you choose to render in the student list so likely the student name